src/inputs.rs

```diff
@@ -232,8 +232,8 @@ impl PartitionedInput {
                                     );
                                     eofd_keys_buffer.insert(key.clone());
```

Comment:
Right now we do trigger this branch of the match statement when the user raises `StopIteration`, and I didn't change that. But yes, since we can now use an empty list to signal that no elements are there _for now_, we can switch to use `return None` instead of `raise StopIteration` for EOF signaling. I'm not entirely sure which way I prefer, raising StopIteration mimicks iterators but returning None feels more natural for control flow. I'm going to make the change, and let's see how it looks, I can always revert the commit if we don't like it.